src/datachain/lib/func.py

```diff
@@ -0,0 +1,50 @@
+from typing import TYPE_CHECKING, Callable, Optional
+
+from sqlalchemy import func as sa_func
+
+from datachain.query.schema import DEFAULT_DELIMITER
+from datachain.sql import functions as dc_func
+
+if TYPE_CHECKING:
+    from datachain import DataType
+
+
+class Func:
```

Comment:
Using custom `Func` class because I can not make it works in all possible cases with default SQLAlchemy functions. Sometimes type detection is not working (for example, for `sum` function) and it fails to run or to store result in DB with correct type.

I will appreciate any comments/recommendations on this 🙏

Reply:
Don't like this part. Decided to revisit this one more last time.

Reply:
Wasn't able to win in this fight with sqlalchemy. I can not find a way to get list of columns with their types from sqlalchemy functions (tried to use [traverse](https://docs.sqlalchemy.org/en/20/core/visitors.html#sqlalchemy.sql.visitors.ExternalTraversal.traverse) but with no luck).